### PR TITLE
fix(conf) check path for plugin commands

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Chekc command path
Fix some exisiting warnings

MON-13295 (21.10, 21.04)
MON-12902 (22.04)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 1]
```diff
@@ -87,14 +94,18 @@
...
+// Exec command only if located in allowed directories
+$dbResult = $pearDB->query('SELECT `resource_line` FROM `cfg_resource`');
+$allowedPaths = $dbResult->fetchAll(\PDO::FETCH_COLUMN);
 
-$stdout = shell_exec($command . " 2>&1");
-$msg = str_replace("\n", "<br />", $stdout);
+$msg = "Command not allowed";
+if (preg_match('#(' . implode('|', $allowedPaths) . ')#', $commandPath)) {
+    $command = $commandPath . ' ' . $plugin . ' ' . $mode . ' --help';
+    $stdout = shell_exec($command . " 2>&1");
```
Comment (callapa) [inline www/include/configuration/configObject/command/minHelpCommand.php:106]: Please protect the command using **escapeshellcmd**

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/include/configuration/configObject/command/minHelpCommand.php (hunk 1 above)
Comment (a-launois) [reply www/include/configuration/configObject/command/minHelpCommand.php:106]: It was already there but I moved it right before the exec

File: www/include/configuration/configObject/command/minHelpCommandFunctions.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+{
+    $allowedResources = getAllResources($pearDB);
+
+    $isAllowed = false;
+    foreach ($allowedResources as $path) {
+        if (substr($command, 0, strlen($path)) === $path) {
+            $isAllowed = true;
+        }
+    }
+    return $isAllowed;
```
Comment (a-launois) [inline www/include/configuration/configObject/command/minHelpCommandFunctions.php:51]:
```suggestion
    foreach ($allowedResources as $path) {
        if (substr($command, 0, strlen($path)) === $path) {
            return true;
        }
    }
    return false;
```

**Review by TamazC — DISMISSED** [review]

File: www/include/configuration/configObject/command/minHelpCommand.php [hunk 3]
```diff
@@ -34,68 +34,48 @@
  *
  */
 
-if (!isset($oreon)) {
+
+if (!isset($centreon)) {
     exit();
 }
 
+require_once $path . "minHelpCommandFunctions.php";
```
Comment (kduret) [inline www/include/configuration/configObject/command/minHelpCommand.php:42]:
```suggestion
require_once __DIR__ . "/minHelpCommandFunctions.php";
```

File: www/include/configuration/configObject/command/minHelpCommandFunctions.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
```
Comment (a-launois) [inline www/include/configuration/configObject/command/minHelpCommandFunctions.php:36]:
```suggestion
declare(strict_types=1);

```

File: www/include/configuration/configObject/command/minHelpCommandFunctions.php [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
```
Comment (a-launois) [inline www/include/configuration/configObject/command/minHelpCommandFunctions.php:36]:
```suggestion

declare(strict_types=1);
```

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**